modules/mtl/arrow-mtl-data/src/test/kotlin/arrow/mtl/KleisliTest.kt

```diff
@@ -3,76 +3,92 @@ package arrow.mtl
 import arrow.core.ForOption
 import arrow.core.ForTry
 import arrow.core.Id
 import arrow.core.Option
 import arrow.core.Try
-import arrow.core.const
 import arrow.core.extensions.`try`.monadError.monadError
 import arrow.core.extensions.const.divisible.divisible
+import arrow.core.extensions.const.eqK.eqK
+import arrow.core.extensions.eq
 import arrow.core.extensions.id.monad.monad
 import arrow.core.extensions.monoid
 import arrow.core.extensions.option.alternative.alternative
+import arrow.core.extensions.option.eqK.eqK
 import arrow.core.some
-import arrow.core.value
 import arrow.fx.ForIO
 import arrow.fx.IO
-import arrow.fx.extensions.io.applicativeError.attempt
 import arrow.fx.extensions.io.concurrent.concurrent
 import arrow.fx.mtl.concurrent
 import arrow.mtl.extensions.kleisli.alternative.alternative
 import arrow.mtl.extensions.kleisli.contravariant.contravariant
 import arrow.mtl.extensions.kleisli.divisible.divisible
 import arrow.mtl.extensions.kleisli.monadError.monadError
 import arrow.test.UnitSpec
+import arrow.test.generators.GenK
+import arrow.test.generators.genK
 import arrow.test.laws.AlternativeLaws
 import arrow.test.laws.ConcurrentLaws
 import arrow.test.laws.ContravariantLaws
 import arrow.test.laws.DivisibleLaws
 import arrow.test.laws.MonadErrorLaws
 import arrow.typeclasses.Conested
 import arrow.typeclasses.Eq
+import arrow.typeclasses.EqK
 import arrow.typeclasses.conest
 import arrow.typeclasses.counnest
+import io.kotlintest.properties.Gen
 import io.kotlintest.shouldBe
 
 class KleisliTest : UnitSpec() {
-  private fun <A> TryEQ(): Eq<KleisliOf<ForTry, Int, A>> = Eq { a, b ->
-    a.run(1) == b.run(1)
-  }
-
-  private fun <A> ConestTryEQ(): Eq<Kind<Conested<Kind<ForKleisli, ForTry>, A>, Int>> = Eq { a, b ->
-    a.counnest().run(1) == b.counnest().run(1)
-  }
 
-  private fun <A> IOEQ(): Eq<Kind<KleisliPartialOf<ForIO, Int>, A>> = Eq { a, b ->
-    a.run(1).attempt().unsafeRunSync() == b.run(1).attempt().unsafeRunSync()
+  private fun conestTryEQK() = object : EqK<Conested<Kind<ForKleisli, ForTry>, Int>> {
+    override fun <A> Kind<Conested<Kind<ForKleisli, ForTry>, Int>, A>.eqK(other: Kind<Conested<Kind<ForKleisli, ForTry>, Int>, A>, EQ: Eq<A>): Boolean {
+      return this.counnest().run(1) == other.counnest().run(1)
+    }
   }
 
   init {
+    fun <F, D> genk(genkF: GenK<F>) = object : GenK<KleisliPartialOf<F, D>> {
+      override fun <A> genK(gen: Gen<A>): Gen<Kind<KleisliPartialOf<F, D>, A>> = genkF.genK(gen).map { k ->
+        Kleisli { _: D -> k }
+      }
+    }
+
+    val optionEQK = Kleisli.eqK(Option.eqK(), Int.eq(), 0)
+
+    val ioEQK = Kleisli.eqK(IO.eqK(), Int.eq(), 1)
+
+    val tryEQK = object : EqK<KleisliPartialOf<ForTry, Int>> {
+      override fun <A> Kind<KleisliPartialOf<ForTry, Int>, A>.eqK(other: Kind<KleisliPartialOf<ForTry, Int>, A>, EQ: Eq<A>): Boolean =
+        (this.fix() to other.fix()).let {
+          it.first.run(1) == it.second.run(1)
+        }
+    }
```

Comment:
If you end up moving the `Try` instance of `EqK` to it's extensions, please rewrite this

Reply:
ok